src/lib/api-client.js

```diff
@@ -205,6 +205,9 @@ export function deleteItem(address){
         itemAddress: address
     }).then(data => {
         Dispatcher.handleAction('ITEMS_RECEIVED', data);
+        Dispatcher.handleAction('ITEM_DELETED', {
```

Comment:
"ITEM_DELETED is only for things that need to know when an item is deleted, which currently is only the toast-store" can also be said for ITEMS_RECEIVED.

The only thing that dispatches an ITEMS_RECEIVED is `GET /model/all`. The fact that `DELETE /item` also returns the entire list is, to be honest, a bit weird and not scalable. It is much more common to only return the deleted elements (see [Javascript][js-splice], [Ruby][rb-delete], [Java][java-remove]). When Tech Support is actually using this with tens of thousands of items, there will most likely be a delay when deleting an item because every single item needs to be enumerated and sent through the network stack. You may say, "Well the user should expect that", but it does not make sense to require loading _all_ items when one is simply deleting an item. When they visit the page that lists all items, it makes intuitive sense that all items will be enumerated because you seeing a list of every single item.

[js-splice]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice
[rb-delete]: http://ruby-doc.org/core-1.9.3/Array.html#method-i-delete
[java-remove]: https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html#remove-java.lang.Object-